src/infer.ts

```diff
@@ -58,26 +61,99 @@ const no_ops: ScalarType = {
 // TODO: https://github.com/hasura/ndc-typescript-deno/issues/21 Use standard logging from SDK
 const LOG_LEVEL = Deno.env.get("LOG_LEVEL") || "INFO";
 const DEBUG = LOG_LEVEL == 'DEBUG';
+const MAX_INFERENCE_RECURSION = 20; // Better to abort than get into an infinite loop, this could be increased if required.
 
-function validate_type(checker: ts.TypeChecker, schema_response: SchemaResponse, name: string, ty: any): Type {
+type TYPE_NAMES = Array<{
```

Comment:
Weird that you named this type in all upper snake case...

Reply:
Yeah I think I started doing this because they were meant to be constants, but some them got ported to arguments. Will rename.

Reply:
c66cefd..b91b2c0